packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.integ.snapshot/asset.bc935fa4a9499a4b71f067e37a7f05ed05bbcdbf4dc33a44747bc8da38db7e92/index.py

```diff
@@ -154,12 +155,15 @@ def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune, ex
             s3_source_zip = s3_source_zips[i]
             markers       = source_markers[i]
 
-            archive=os.path.join(workdir, str(uuid4()))
-            logger.info("archive: %s" % archive)
-            aws_command("s3", "cp", s3_source_zip, archive)
-            logger.info("| extracting archive to: %s\n" % contents_dir)
-            logger.info("| markers: %s" % markers)
-            extract_and_replace_markers(archive, contents_dir, markers)
+            if extract:
+                archive=os.path.join(workdir, str(uuid4()))
+                logger.info("archive: %s" % archive)
+                aws_command("s3", "cp", s3_source_zip, archive)
+                logger.info("| extracting archive to: %s\n" % contents_dir)
+                logger.info("| markers: %s" % markers)
+                extract_and_replace_markers(archive, contents_dir, markers)
+            else:
+                aws_command("s3", "cp", s3_source_zip, contents_dir)
```

Comment:
Return an output property that contains the full generated object key (excluding the bucket name). This is going to entail adding a field to the `responseData` object here:

https://github.com/aws/aws-cdk/blob/1f03e8c40a682a3b5aae90560c84017cfe62762e/packages/%40aws-cdk/aws-s3-deployment/lib/lambda/index.py#L121

And making it accessible as a property via a `this.cr.getAtt()` like here:

https://github.com/aws/aws-cdk/blob/1f03e8c40a682a3b5aae90560c84017cfe62762e/packages/%40aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts#L433

It can be in a getter that returns a string. Don't forget to document it.

Reply:
There are multiple object keys if multiple sources are provided to the `BucketDeployment` construct. Would it make the most sense to return a list of strings which are the object keys?

Reply:
Yes, that makes sense